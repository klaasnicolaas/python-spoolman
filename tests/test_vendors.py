"""Test the Vendor model for Spoolman."""

from __future__ import annotations

from typing import TYPE_CHECKING

from aresponses import ResponsesMockServer
from syrupy.assertion import SnapshotAssertion

from . import load_fixtures

if TYPE_CHECKING:
    from spoolman import Spoolman


async def test_get_vendors(
    aresponses: ResponsesMockServer,
    spoolman_client: Spoolman,
    snapshot: SnapshotAssertion,
) -> None:
    """Test getting a list of vendors."""
    aresponses.add(
        "192.168.x.x:7912",
        "/api/v1/vendor",
        "GET",
        aresponses.Response(
            status=200,
            headers={"Content-Type": "application/json"},
            text=load_fixtures("vendors.json"),
        ),
    )
    vendors = await spoolman_client.get_vendors()
    assert vendors == snapshot


async def test_get_vendor(
    aresponses: ResponsesMockServer,
    spoolman_client: Spoolman,
    snapshot: SnapshotAssertion,
) -> None:
    """Test getting a specific vendor."""
    aresponses.add(
        "192.168.x.x:7912",
        "/api/v1/vendor/1",
        "GET",
        aresponses.Response(
            status=200,
            headers={"Content-Type": "application/json"},
            text=load_fixtures("single_vendor.json"),
        ),
    )
    vendor = await spoolman_client.get_vendor(vendor_id=1)
    assert vendor == snapshot
