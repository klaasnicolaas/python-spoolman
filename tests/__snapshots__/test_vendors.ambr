# serializer version: 1
# name: test_get_vendor
  Vendor(id=1, name='Bambu Lab', external_id='Bambu Lab', registered=datetime.datetime(2024, 12, 28, 23, 32, 3, tzinfo=datetime.timezone.utc))
# ---
# name: test_get_vendors
  list([
    Vendor(id=1, name='Bambu Lab', external_id='Bambu Lab', registered=datetime.datetime(2024, 12, 28, 23, 32, 3, tzinfo=datetime.timezone.utc)),
    Vendor(id=2, name='Polymaker', external_id='Polymaker', registered=datetime.datetime(2024, 12, 28, 23, 32, 22, tzinfo=datetime.timezone.utc)),
  ])
# ---
